library/blob.c

```diff
@@ -489,6 +527,29 @@ int eblob_check_record(const struct eblob_base_ctl *bctl,
 		return -ESPIPE;
 	}
 
+	/*
+	 * If there is no no-checksum bit, there must be enough space in the footer for checksum.
+	 */
+	if (!(dc->flags & BLOB_DISK_CTL_NOCSUM)) {
+		long footer_min_size = sizeof(struct eblob_disk_footer);
+		if (dc->flags & BLOB_DISK_CTL_CHUNKED_CSUM) {
+			footer_min_size = 0;
+
+			if (dc->data_size)
+				footer_min_size = ((dc->data_size - 1) / EBLOB_CSUM_CHUNK_SIZE + 1) * sizeof(uint64_t);
```

Comment:
Nope, it returns zero if config sets no-footer flag, but we should check on-disk sizes regardless of current blob settings.